File: api/src/main/java/dev/ulloasp/mlsuite/role/application/dto/RoleDefinitionDto.java
package dev.ulloasp.mlsuite.role.application.dto;

import java.util.List;

public record RoleDefinitionDto(
        Long id,
        String name,
        String slug,
        String description,
        String scope,
        boolean locked,
        String systemKey,
        long userCount,
        List<PermissionDto> permissions,
        RoleActionsDto actions) {
}

File: frontend/src/api/workspace/dtos/role-definition-dto.ts
/*
SPDX-License-Identifier: MIT
Copyright (c) 2025 Pablo Ulloa Santin
*/

import type { PermissionDto, RoleScope } from "./index";

export interface RoleDefinitionDto {
  id: number;
  name: string;
  slug: string;
  description: string;
  scope: RoleScope;
  locked: boolean;
  systemKey?: string | null;
  userCount: number;
  permissions: PermissionDto[];
  actions: {
    canView: boolean;
    canEdit: boolean;
    canDelete: boolean;
    canDuplicate: boolean;
    canAssign: boolean;
  };
}
